lib/readline.js

```diff
@@ -456,6 +458,28 @@ Interface.prototype._insertString = function(c) {
 
     // a hack to get the line refreshed if it's needed
     this._moveCursor(0);
+    // emit current line for generating preview
+    this.emit('buffer', this.line);
```

Comment:
This line is there in both the branches, at the end. Perhaps this can be moved out of the `if..else`.